Sources/Testcontainers/GenericContainer.swift

```diff
@@ -5,34 +5,35 @@ import NIO
 
 public final class GenericContainer {
     
-    let logger = Logger(label: "GenericContainer")
-    
     static let uuid = UUID().uuidString
     
-    private let name: String
+    private let logger: Logger
+    private let imageParams: ImageParams
     private let configuration: ContainerConfig
     
     private var docker: Docker?
     private var container: Docker.Container?
     private var image: Docker.Image?
     
-    public init(name: String, configuration: ContainerConfig) {
-        self.name = name
+    public init(image: ImageParams, configuration: ContainerConfig, logger: Logger? = nil) {
+        self.imageParams = image
         self.configuration = configuration
+        self.logger = logger ?? Logger(label: "GenericContainer")
```

Comment:
What we like to do in our Swift apps is use a default initialize for the Logger to make things a bit cleaner, like this:

```swift
    logger: Logger = Logger(label: String(describing: GenericContainer.self))
```

But, this is just a suggestion.